src/hotspot/share/opto/superword.cpp

```diff
@@ -1316,7 +1316,16 @@ bool SuperWord::have_similar_inputs(Node* s1, Node* s2) {
   // assert(independent(s1, s2) == true, "check independent");
   if (s1->req() > 1 && !s1->is_Store() && !s1->is_Load()) {
     for (uint i = 1; i < s1->req(); i++) {
-      if (s1->in(i)->Opcode() != s2->in(i)->Opcode()) return false;
+      Node* s1_in = s1->in(i);
+      Node* s2_in = s2->in(i);
+      if (s1_in->is_Phi() && s2_in->is_Add() && s2_in->in(1) == s1_in) {
+        // Special handling for expressions with loop iv, like "b[i] = a[i] * i".
+        // In this case, one node has an input from the tripcount iv and another
+        // node has an input from iv plus an offset.
+        if (!s1_in->as_Phi()->is_tripcount(T_INT)) return false;
+      } else {
+        if (s1_in->Opcode() != s1_in->Opcode()) return false;
```

Comment:
This last line ought to be `s1_in->Opcode() != s2_in->Opcode` but you have replaced `s2` with `s1`.